index.bs

```diff
@@ -1370,77 +896,768 @@ What configuration is the device in after it resets?
+    steps <a>in parallel</a>:
+    1. <a>Check if the device is configured</a> with <a>this</a> and |promise|
+        and abort these steps if promise is rejected.
+    1. Let |endpointAddress| be <code>|endpointNumber| | 0x80</code>
+        (i.e {{USBDirection/"in"}} direction).
+    1. Let |endpoint| be the result of <a>finding the endpoint</a> with
+        |endpointAddress| and <a>this</a>.
+    1. If |endpoint| is <code>null</code>, <a>reject</a> |promise| with a
+        {{NotFoundError}} and abort these steps.
+    1. If |endpoint|.{{USBEndpoint/type}} is not equal to {{USBEndpointType/"isochronous"}},
+        <a>reject</a> |promise| with an {{InvalidAccessError}} and abort these steps.
+    1. Let |length| be the sum of the elements of |packetLengths|.
+    1. Let |buffer| be a new {{ArrayBuffer}} of |length| bytes.
+    1. Let |result| be a new {{USBIsochronousInTransferResult}} and set
+        <code>|result|.{{USBIsochronousInTransferResult/data}}</code> to a new
+        {{DataView}} constructed over |buffer|.
+    1. Enqueue an <a lt="isochronous transfers">isochronous IN transfer</a> on
+        |endpoint| that will write up to |length| bytes of data from the device
+        into |buffer|.
+    1. For each packet |i| from <code>0</code> to <code>|packetLengths|.length -
+        1</code>:
+        1. Let |packet| be a new {{USBIsochronousInTransferPacket}} and set
+            <code>|result|.{{USBIsochronousInTransferResult/packets}}[|i|]</code>
+            to |packet|.
+        1. Let |view| be a new {{DataView}} over the portion of |buffer|
+            containing the data received from the device for this packet and set
+            <code>|packet|.{{USBIsochronousInTransferPacket/data}}</code> to
+            |view|.
+        1. If the device responds with more than <code>|packetLengths|[i]</code>
+            bytes of data set
+            <code>|packet|.{{USBIsochronousInTransferPacket/status}}</code> to
+            {{"babble"}}.
+        1. If the transfer ends because |endpoint| is stalled set
+            <code>|packet|.{{USBIsochronousInTransferPacket/status}}</code> to
+            {{"stall"}}.
+        1. If the device acknowledges the complete transfer set
+            <code>|packet|.{{USBIsochronousInTransferPacket/status}}</code> to
+            {{"ok"}}.
+        1. If the transfer fails for any other reason <a>reject</a> |promise|
+            with a {{NetworkError}} and abort these steps.
+    1. <a>Resolve</a> |promise| with |result|.
+</div>
+
+<div algorithm="USBDevice.isochronousTransferOut(endpointNumber, data, packetLengths)">
+    The <dfn for=USBDevice method>isochronousTransferOut(|endpointNumber|, |data|, |packetLengths|)</dfn> method,
+    when invoked, MUST return a new {{Promise}} |promise| and run the
+    following steps <a>in parallel</a>:
+    1. <a>Check if the device is configured</a> with <a>this</a> and |promise|
+        and abort these steps if promise is rejected.
```

Comment:
```suggestion
        and abort these steps if |promise| is rejected.
```

Reply:
Updated in https://github.com/WICG/webusb/pull/212/commits/5aaf696938ed1dc073f8cc8da8c809e9e5019f41.